modules/abstract-cosmos/src/lib/transactionBuilder.ts

```diff
@@ -128,4 +135,122 @@ export abstract class CosmosTransactionBuilder extends BaseTransactionBuilder {
       throw new BuildTransactionError(`Key validation failed`);
     }
   }
+
+  /**
+   * Sets gas budget of this transaction
+   * Gas budget consist of fee amount and gas limit. Division feeAmount/gasLimit represents
+   * the gas-fee and it should be more than minimum required gas-fee to process the transaction
+   * @param {FeeData} gasBudget
+   * @returns {TransactionBuilder} this transaction builder
+   */
+  gasBudget(gasBudget: FeeData): this {
```

Comment:
```suggestion
  setGasBudget(gasBudget: FeeData): this {
```

Reply:
Other setter methods are also not using set prefix. Will take care of this in a separate PR.

Reply:
Kindly add a todo ticket for this